src/epcpm/anomaliestoxlsx.py

```diff
@@ -0,0 +1,275 @@
+import attr
+import openpyxl
+import typing
+
+import epyqlib.pm.parametermodel
+import epyqlib.utils.general
+
+import epcpm.c
+import epcpm.pm_helper
+import epcpm.anomalymodel
+
+builders = epyqlib.utils.general.TypeMap()
+
+
+@attr.s
+class Fields(epcpm.pm_helper.FieldsInterface):
+    """The fields defined for a given row in the output XLS file."""
+
+    name = attr.ib(default=None, type=typing.Union[str, bool])
+    group = attr.ib(default=None, type=typing.Union[str, bool])
+    code = attr.ib(default=None, type=typing.Union[int, bool])
+    trig_type = attr.ib(default=None, type=typing.Union[int, bool])
+    resp_level_I = attr.ib(default=None, type=typing.Union[int, bool])
+    resp_level_A = attr.ib(default=None, type=typing.Union[int, bool])
+    comment = attr.ib(default=None, type=typing.Union[str, bool])
+
+
+field_names = Fields(
+    name="Name",
+    group="Group",
+    code="Code",
+    trig_type="Trigger type",
+    resp_level_I="Response level I",
+    resp_level_A="Response level A",
+    comment="Description",
+)
+
+
+@attr.s
+class InfoSheetFields(epcpm.pm_helper.FieldsInterface):
+    """The fields defined for response level information sheet."""
+
+    name = attr.ib(default=None, type=typing.Union[str, bool])
+    desc = attr.ib(default=None, type=typing.Union[str, bool])
+
+
+info_sheet_field_names = InfoSheetFields(
+    name="Name",
+    desc="Description",
+)
+
+"""
+Exports anomaly information to a spreadsheet file.
+
+Args:
+    path:             Path to output XLSX file.
+    anomaly_model:    Anomaly data model.
+    parameters_model: Parameter data model.
+    column_filter:    Optional filter to remove selected output columns.
+    skip_output:      Optional boolean flag to disable output.
+
+Returns:
+    None
+"""
+
+
+def export(
+    path,
+    anomaly_model,
+    parameters_model,
+    column_filter=None,
+    skip_output=False,
+):
```

Comment:
The style is to put docstrings below the `def`. I only put a review comment on this one, but there are others throughout this PR.

```suggestion

def export(
    path,
    anomaly_model,
    parameters_model,
    column_filter=None,
    skip_output=False,
):
    """
    Exports anomaly information to a spreadsheet file.

    Args:
        path:             Path to output XLSX file.
        anomaly_model:    Anomaly data model.
        parameters_model: Parameter data model.
        column_filter:    Optional filter to remove selected output columns.
        skip_output:      Optional boolean flag to disable output.

    Returns:
        None
    """

```